hledger-lib/Hledger/Utils/String.hs

```diff
@@ -134,10 +134,10 @@ whitespacechars = " \t\n\r"
 redirectchars   = "<>"
 
 escapeDoubleQuotes :: String -> String
-escapeDoubleQuotes = regexReplace "\"" "\""
+escapeDoubleQuotes = id  -- regexReplace "\"" "\""
```

Comment:
Actually, I think this is a no-op in any case. The original code isn't actually escaping anything, just replacing every quotation mark with another quotation mark. This should either be fixed if it is supposed to do something, or be removed.

Reply:
You're right! String.escapeDoubleQuotes never worked. Committed in 2014 with message: "More crazy quote engineering.. seems more robust than before."

Text.escapeDoubleQuotes was fixed in 2018:
```haskell
escapeDoubleQuotes = T.replace "\"" "\\\""
```
but not the original String version.

It's not exported, so might as well be moved into quoteIfNeeded, its only user. 

Feel free to fix it, or ignore it and I'll do after this PR.